torch/csrc/autograd/python_variable.cpp

```diff
@@ -188,42 +188,18 @@ void pushPyOutToStack(
 
 namespace {
 
-template <const char* func_name, typename... Ts>
-void concrete_trace_cuda(Ts... args) {
-  pybind11::gil_scoped_acquire gil;
-  at::impl::MaybeSetTLSOnEntryGuard guard;
-
-  if (Py_IsInitialized()) {
-    try {
-      py::module mod = py::module::import("torch.utils._cuda_trace");
-      py::object hook = mod.attr(func_name).attr("fire_callbacks");
-      hook(args...);
-    } catch (const std::exception& e) {
-      LOG(ERROR) << "CUDA trace hook execution failed: " << e.what();
-    }
+#define CONCRETE_TRACE_CUDA(func_name, ...) \
+  pybind11::gil_scoped_acquire gil; \
```

Comment:
I know it was there before but can you move this into the `Py_IsInitialized()` scope? That might prevent some deadlock.